src/transformers/clip.rs

```diff
@@ -346,47 +327,48 @@ pub struct ClipTextTransformer<B: Backend> {
 
 impl<B: Backend> ClipTextTransformer<B> {
     // https://github.com/huggingface/transformers/blob/674f750a57431222fa2832503a108df3badf1564/src/transformers/models/clip/modeling_clip.py#L678
-    fn build_causal_attention_mask(bsz: usize, seq_len: usize, device: &B::Device) -> Tensor<B, 2> {
-        let mask = Tensor::full_device([bsz, seq_len, seq_len], f32::MIN, device);
-        let mask = zero_lower_diagonal(mask); // zero out the lower diagonal
-                                              // expand mask
-        mask.unsqueeze()
+    fn build_causal_attention_mask(bsz: usize, seq_len: usize, device: &B::Device) -> Tensor<B, 4> {
+        let mut mask_vec = Vec::with_capacity(seq_len * seq_len);
```

Comment:
maybe we could move that to a utils module

Reply:
I did, and will merge now!